src/BitmapText.cpp

```diff
@@ -672,8 +686,45 @@ bool BitmapText::EarlyAbortDraw() const
 	return m_wTextLines.empty();
 }
 
+namespace {
+	// Helper functions used in DrawPrimitives.
+	inline void SetVertexColor(std::vector<RageSpriteVertex>& vertices, size_t i, const RageColor* color)
+	{
+		vertices[i + 0].c = color[0]; // top left
+		vertices[i + 1].c = color[2]; // bottom left
+		vertices[i + 2].c = color[3]; // bottom right
+		vertices[i + 3].c = color[1]; // top right
+	}
+
+	inline void ApplyJitter(std::vector<RageSpriteVertex>& vertices, size_t i, const RageVector3& jitter)
+	{
+		vertices[i + 0].p += jitter; // top left
+		vertices[i + 1].p += jitter; // bottom left
+		vertices[i + 2].p += jitter; // bottom right
+		vertices[i + 3].p += jitter; // top right
+	}
+
+	inline void UndoJitter(std::vector<RageSpriteVertex>& vertices, size_t i, const RageVector3& jitter)
+	{
+		vertices[i + 0].p -= jitter; // top left
+		vertices[i + 1].p -= jitter; // bottom left
+		vertices[i + 2].p -= jitter; // bottom right
+		vertices[i + 3].p -= jitter; // top right
+	}
+
+	constexpr size_t ClampToVertexSize(size_t iEnd, size_t verticesSize)
+	{
+		return (iEnd < verticesSize) ? iEnd : verticesSize;
```

Comment:
Why not just std::min?

Reply:
The real reason was that I thought `ClampToVertexSize` is much easier for the reader to understand _why_ the minimum is being determined - there's no actual benefit here to using `std::min` outside of readability. So I don't mind going back to `std::min` and maybe adding a comment saying `// clamp to vertex size` or something if this seems messy to you.